module/zfs/metaslab.c

```diff
@@ -1221,6 +1337,131 @@ metaslab_rangesize_compare(const void *x1, const void *x2)
 
 	return (AVL_CMP(r1->rs_start, r2->rs_start));
 }
+typedef struct metaslab_rt_arg {
+	btree_t *mra_bt;
+	uint32_t mra_floor_shift;
+} metaslab_rt_arg_t;
+
+struct mssa_arg {
+	range_tree_t *rt;
+	metaslab_rt_arg_t *mra;
+};
+
+static void
+metaslab_size_sorted_add(void *arg, uint64_t start, uint64_t size)
+{
+	struct mssa_arg *mssap = arg;
+	range_tree_t *rt = mssap->rt;
+	metaslab_rt_arg_t *mrap = mssap->mra;
+	range_seg_max_t seg = {0};
+	rs_set_start(&seg, rt, start);
+	rs_set_end(&seg, rt, start + size);
+	metaslab_rt_add(rt, &seg, mrap);
+}
+
+static void
+metaslab_size_tree_full_load(range_tree_t *rt)
+{
+	metaslab_rt_arg_t *mrap = rt->rt_arg;
+	btree_t *t = mrap->mra_bt;
```

Comment:
For non-debug builds, `zfs/module/zfs/metaslab.c:1366:11: warning: unused variable ‘t’ [-Wunused-variable]`.  Let's simply use `mrap->mra_bt` in the ASSERT.